Checkout/Checkout/Source/Model/Card+Scheme.swift

```diff
@@ -133,7 +138,43 @@ extension Card {
         return [3]
       }
     }
-
+      
+    public var rawValue: String {
+        switch self {
+        case .unknown:
+            return "unknown"
+        case .mada:
+            return "mada"
+        case .visa:
+            return "visa"
+        case .mastercard:
+            return "mastercard"
+        case .maestro:
+            return "maestro"
+        case .americanExpress:
+            return "americanExpress"
+        case .discover:
+            return "discover"
+        case .dinersClub:
+            return "dinersClub"
+        case .jcb:
+            return "jcb"
+        }
+    }
+      
+    /// map of card scheme to indexes of spaces in formatted card number string
+    /// eg. a visa card has gaps at 4, 8 and 12. 4242424242424242 becomes 4242 4242 4242 4242
+    public var cardGaps: [Int] {
+        switch self {
+        case .unknown:
+            return []
+        case .mada, .visa, .mastercard, .discover, .maestro, .jcb:
+            return [4, 8, 12]
```

Comment:
Should we add card gap for more than 16?

Reply:
@patrick-hoban-cko has had a look at that and said we'd do it separately as he wasn't sure. He was actually planning to ask @harry-brown-cko , so maybe we can sort it out here.

Reply:
[check out this blog for some examples](https://baymard.com/checkout-usability/credit-card-patterns)

I think we can start by using `[4, 8, 12, 16]` for these cards and fix specific formatting issues if they are raised by merchants. @ehab-al-cko @alex-ioja-yang-cko thoughts?

Reply:
When I spoke to @patrick-hoban-cko on it his opinion was to ask you @harry-brown-cko . Since you and @ehab-al-cko obviously also feel the space could be there, I have submitted it now as part of this PR 👍